### ISSUE TITLE
azure key-vault can't take multiple powershell scripts connecting to take secret value with

### ISSUE THREAD

**Issue by konstantinmds** [Azure/azure-powershell#9739]
When executing single powershell script, i get credentials from my key vault, or when i run them sequentialy ( on the same key vault secret).

I connect like this : 
```
(Get-AzureKeyVaultSecret -vaultName $vault_name -Name $name).SecretValueText
```

But when i run scripts in parallel, where every one of them needs to to get the secret from the vault, i get following error :
" Get-AzureKeyVaultSecret : user_interaction_required: One of two conditions was encountered: 1. The PromptBehavior.Never flag was passed, but the constraint could not be honored, because user interaction was required. 2. An error occurred during a silent web authentication that prevented the http authentication flow from completing in a short enough time frame "

It seems that problem is on silent web authentication. And it works on five subsequent times,
and it fails on the next time on the same script.
I tried to sleep for a second and retry connection, but without much success.

**Comment by niander** [comment]
@konstantinmds Hi. Could you give us the debug output? Please run `$DebugPreference = "Continue"` before running the above commands.

**Comment by konstantinmds** [comment]
@niander Here it is :

`
Microsoft.IdentityModel.Clients.ActiveDirectory.AdalException: user_interaction_required: One of two conditions was 

encountered: 1. The PromptBehavior.Never flag was passed, but the constraint could not be honored, because user 

interaction was required. 2. An error occurred during a silent web authentication that prevented the http 

authentication flow from completing in a short enough time frame



   at Microsoft.IdentityModel.Clients.ActiveDirectory.AcquireTokenInteractiveHandler.VerifyAuthorizationResult()



   at Microsoft.IdentityModel.Clients.ActiveDirectory.AcquireTokenInteractiveHandler.PreTokenRequest()



   at Microsoft.IdentityModel.Clients.ActiveDirectory.AcquireTokenHandlerBase.<RunAsync>d__0.MoveNext()

 ErrorCode: user_interaction_required

DEBUG: AzureQoSEvent: CommandName - Get-AzureKeyVaultSecret; IsSuccess - False; Duration - 00:01:22.6863338; Exception 

- Microsoft.IdentityModel.Clients.ActiveDirectory.AdalException: user_interaction_required: One of two conditions was 

encountered: 1. The PromptBehavior.Never flag was passed, but the constraint could not be honored, because user 

interaction was required. 2. An error occurred during a silent web authentication that prevented the http 

authentication flow from completing in a short enough time frame



   at Microsoft.Azure.Commands.KeyVault.Models.KeyVaultDataServiceClient.GetSecret(String vaultName, String secretName,

 String secretVersion)



   at Microsoft.Azure.Commands.KeyVault.GetAzureKeyVaultSecret.ExecuteCmdlet()



   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()

 ErrorCode: user_interaction_required;

DEBUG: Finish sending metric.

DEBUG: 6:16:25 PM - GetAzureKeyVaultSecret end processing.

DEBUG: 6:16:27 PM - GetAzureKeyVaultSecret end processing.

-1
`

**Comment by konstantinmds** [comment]
Probably the thing can be using of the AzureRM module with AzModule.

**Comment by niander** [comment]
Could you list here all modules you have in your system? `Get-Module -ListAvailable`

**Comment by konstantinmds** [comment]
Output : 
ModuleType Version    Name                                ExportedCommands       
---------- -------    ----                                ----------------       
Script     1.6.0      Az.Accounts                         {Disable-AzDataColle...
Script     1.0.0      Az.Advisor                          {Get-AzAdvisorRecomm...
Script     1.0.1      Az.Aks                              {Get-AzAks, New-AzAk...
Script     1.1.0      Az.AnalysisServices                 {Resume-AzAnalysisSe...
Script     1.2.0      Az.ApiManagement                    {Add-AzApiManagement...
Script     1.0.0      Az.ApplicationInsights              {Get-AzApplicationIn...
Script     1.3.0      Az.Automation                       {Get-AzAutomationHyb...
Script     1.1.0      Az.Batch                            {Remove-AzBatchAccou...
Script     1.0.0      Az.Billing                          {Get-AzBillingInvoic...
Script     1.3.0      Az.Cdn                              {Get-AzCdnProfile, G...
Script     1.1.1      Az.CognitiveServices                {Get-AzCognitiveServ...
Script     2.4.0      Az.Compute                          {Remove-AzAvailabili...
Script     1.0.1      Az.ContainerInstance                {New-AzContainerGrou...
Script     1.0.1      Az.ContainerRegistry                {New-AzContainerRegi...
Script     1.1.2      Az.DataFactory                      {Set-AzDataFactoryV2...
Script     1.0.0      Az.DataLakeAnalytics                {Get-AzDataLakeAnaly...
Script     1.2.1      Az.DataLakeStore                    {Get-AzDataLakeStore...
Script     1.0.0      Az.DeploymentManager                {Get-AzDeploymentMan...
Script     1.0.0      Az.DevTestLabs                      {Get-AzDtlAllowedVMS...
Script     1.1.1      Az.Dns                              {Get-AzDnsRecordSet,...
Script     1.2.1      Az.EventGrid                        {New-AzEventGridTopi...
Script     1.2.0      Az.EventHub                         {New-AzEventHubNames...
Script     1.1.0      Az.FrontDoor                        {New-AzFrontDoor, Ge...
Script     2.0.0      Az.HDInsight                        {Get-AzHDInsightJob,...
Script     1.2.0      Az.IotHub                           {Add-AzIotHubKey, Ge...
Script     1.2.0      Az.KeyVault                         {Add-AzKeyVaultCerti...
Script     1.2.1      Az.LogicApp                         {Get-AzIntegrationAc...
Script     1.1.0      Az.MachineLearning                  {Move-AzMlCommitment...
Script     1.0.0      Az.MarketplaceOrdering              {Get-AzMarketplaceTe...
Script     1.1.0      Az.Media                            {Sync-AzMediaService...
Script     1.2.1      Az.Monitor                          {Get-AzMetricDefinit...
Script     1.11.0     Az.Network                          {Add-AzApplicationGa...
Script     1.1.0      Az.NotificationHubs                 {Get-AzNotificationH...
Script     1.3.1      Az.OperationalInsights              {New-AzOperationalIn...
Script     1.1.2      Az.PolicyInsights                   {Get-AzPolicyEvent, ...
Script     1.1.0      Az.PowerBIEmbedded                  {Remove-AzPowerBIWor...
Script     1.4.2      Az.RecoveryServices                 {Get-AzRecoveryServi...
Script     1.1.0      Az.RedisCache                       {Remove-AzRedisCache...
Script     1.0.1      Az.Relay                            {New-AzRelayNamespac...
Script     1.6.0      Az.Resources                        {Get-AzProviderOpera...
Script     1.2.1      Az.ServiceBus                       {New-AzServiceBusNam...
Script     1.1.1      Az.ServiceFabric                    {Add-AzServiceFabric...
Script     1.0.2      Az.SignalR                          {New-AzSignalR, Get-...
Script     1.13.0     Az.Sql                              {Get-AzSqlDatabaseTr...
Script     1.5.0      Az.Storage                          {Get-AzStorageAccoun...
Script     1.1.1      Az.StorageSync                      {Invoke-AzStorageSyn...
Script     1.0.0      Az.StreamAnalytics                  {Get-AzStreamAnalyti...
Script     1.0.1      Az.TrafficManager                   {Add-AzTrafficManage...
Script     1.3.0      Az.Websites                         {Get-AzAppServicePla...
Script     0.5.4      Azure.AnalysisServices              {Add-AzureAnalysisSe...
Script     4.6.1      Azure.Storage                       {Get-AzureStorageTab...
Binary     2.0.2.26   AzureAD                             {Add-AzureADApplicat...
Script     6.13.1     AzureRM                                                    
Script     0.6.14     AzureRM.AnalysisServices            {Resume-AzureRmAnaly...
Script     6.1.7      AzureRM.ApiManagement               {Add-AzureRmApiManag...
Script     0.1.8      AzureRM.ApplicationInsights         {Get-AzureRmApplicat...
Script     6.1.1      AzureRM.Automation                  {Get-AzureRMAutomati...
Script     4.0.11     AzureRM.Backup                      {Backup-AzureRmBacku...
Script     4.1.5      AzureRM.Batch                       {Remove-AzureRmBatch...
Script     0.14.6     AzureRM.Billing                     {Get-AzureRmBillingI...
Script     5.0.6      AzureRM.Cdn                         {Get-AzureRmCdnProfi...
Script     0.9.12     AzureRM.CognitiveServices           {Get-AzureRmCognitiv...
Script     5.9.1      AzureRM.Compute                     {Remove-AzureRmAvail...
Script     0.3.7      AzureRM.Consumption                 {Get-AzureRmConsumpt...
Script     0.2.12     AzureRM.ContainerInstance           {New-AzureRmContaine...
Script     1.0.10     AzureRM.ContainerRegistry           {New-AzureRmContaine...
Script     5.0.3      AzureRM.DataFactories               {Remove-AzureRmDataF...
Script     0.5.11     AzureRM.DataFactoryV2               {Set-AzureRmDataFact...
Script     5.1.4      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLake...
Script     6.2.1      AzureRM.DataLakeStore               {Get-AzureRmDataLake...
Script     4.0.9      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllow...
Script     5.1.0      AzureRM.Dns                         {Get-AzureRmDnsRecor...
Script     0.3.7      AzureRM.EventGrid                   {New-AzureRmEventGri...
Script     0.7.0      AzureRM.EventHub                    {New-AzureRmEventHub...
Script     4.1.8      AzureRM.HDInsight                   {Get-AzureRmHDInsigh...
Script     5.1.5      AzureRM.Insights                    {Get-AzureRmMetricDe...
Script     3.1.8      AzureRM.IotHub                      {Add-AzureRmIotHubKe...
Script     5.2.1      AzureRM.KeyVault                    {Add-AzureKeyVaultCe...
Script     4.1.4      AzureRM.LogicApp                    {Get-AzureRmIntegrat...
Script     0.18.5     AzureRM.MachineLearning             {Move-AzureRmMlCommi...
Script     0.4.8      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpClus...
Script     0.2.7      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketpl...
Script     0.10.4     AzureRM.Media                       {Sync-AzureRmMediaSe...
Script     6.11.1     AzureRM.Network                     {Add-AzureRmApplicat...
Script     5.0.3      AzureRM.NotificationHubs            {Get-AzureRmNotifica...
Script     5.0.6      AzureRM.OperationalInsights         {New-AzureRmOperatio...
Script     1.1.0      AzureRM.PolicyInsights              {Get-AzureRmPolicyEv...
Script     4.1.10     AzureRM.PowerBIEmbedded             {Remove-AzureRmPower...
Script     5.8.2      AzureRM.profile                     {Disable-AzureRmData...
Script     4.1.9      AzureRM.RecoveryServices            {Get-AzureRmRecovery...
Script     4.5.2      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecov...
Script     0.2.12     AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecover...
Script     5.1.0      AzureRM.RedisCache                  {Remove-AzureRmRedis...
Script     0.3.12     AzureRM.Relay                       {New-AzureRmRelayNam...
Script     6.7.3      AzureRM.Resources                   {Get-AzureRmProvider...
Script     0.16.10    AzureRM.Scheduler                   {Disable-AzureRmSche...
Script     0.6.13     AzureRM.ServiceBus                  {New-AzureRmServiceB...
Script     0.3.15     AzureRM.ServiceFabric               {Add-AzureRmServiceF...
Script     1.0.0      AzureRM.SignalR                     {New-AzureRmSignalR,...
Script     4.12.1     AzureRM.Sql                         {Get-AzureRmSqlDatab...
Script     5.2.0      AzureRM.Storage                     {Get-AzureRmStorageA...
Script     4.0.10     AzureRM.StreamAnalytics             {Get-AzureRmStreamAn...
Script     4.0.5      AzureRM.Tags                        {Remove-AzureRmTag, ...
Script     4.1.3      AzureRM.TrafficManager              {Add-AzureRmTrafficM...
Script     4.0.5      AzureRM.UsageAggregates             Get-UsageAggregates    
Script     5.2.0      AzureRM.Websites                    {Get-AzureRmAppServi...
Manifest   1.1.183.17 MSOnline                            {Get-MsolDevice, Rem...


    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands       
---------- -------    ----                                ----------------       
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValida...
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-P...
Script     3.4.0      Pester                              {Describe, Context, ...
Script     1.0.0.1    PowerShellGet                       {Install-Module, Fin...
Script     2.0.0      PSReadline                          {Get-PSReadLineKeyHa...


    Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules


ModuleType Version    Name                                ExportedCommands       
---------- -------    ----                                ----------------       
Manifest   1.0.0.0    AppBackgroundTask                   {Disable-AppBackgrou...
Manifest   2.0.0.0    AppLocker                           {Get-AppLockerFileIn...
Manifest   1.0.0.0    AppvClient                          {Add-AppvClientConne...
Manifest   2.0.1.0    Appx                                {Add-AppxPackage, Ge...
Script     1.0.0.0    AssignedAccess                      {Clear-AssignedAcces...
Manifest   1.0.0.0    BitLocker                           {Unlock-BitLocker, S...
Manifest   2.0.0.0    BitsTransfer                        {Add-BitsFile, Compl...
Manifest   1.0.0.0    BranchCache                         {Add-BCDataCacheExte...
Manifest   1.0.0.0    CimCmdlets                          {Get-CimAssociatedIn...
Manifest   1.0        ConfigCI                            {Get-SystemDriver, N...
Manifest   1.0        Defender                            {Get-MpPreference, S...
Manifest   1.0.2.0    DeliveryOptimization                {Delete-DeliveryOpti...
Manifest   1.0.0.0    DirectAccessClientComponents        {Disable-DAManualEnt...
Script     3.0        Dism                                {Add-AppxProvisioned...
Manifest   1.0.0.0    DnsClient                           {Resolve-DnsName, Cl...
Manifest   1.0.0.0    EventTracingManagement              {Start-EtwTraceSessi...
Manifest   2.0.0.0    International                       {Get-WinDefaultInput...
Manifest   1.0.0.0    iSCSI                               {Get-IscsiTargetPort...
Script     1.0.0.0    ISE                                 {New-IseSnippet, Imp...
Manifest   1.0.0.0    Kds                                 {Add-KdsRootKey, Get...
Manifest   1.0.1.0    Microsoft.PowerShell.Archive        {Compress-Archive, E...
Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    {Get-WinEvent, Get-C...
Manifest   3.0.0.0    Microsoft.PowerShell.Host           {Start-Transcript, S...
Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  {Add-LocalGroupMembe...
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-...
Script     1.0        Microsoft.PowerShell.ODataUtils     Export-ODataEndpoint...
Manifest   3.0.0.0    Microsoft.PowerShell.Security       {Get-Acl, Set-Acl, G...
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Format-List, Format...
Manifest   3.0.0.0    Microsoft.WSMan.Management          {Disable-WSManCredSS...
Manifest   1.0        MMAgent                             {Disable-MMAgent, En...
Manifest   1.0.0.0    MsDtc                               {New-DtcDiagnosticTr...
Manifest   2.0.0.0    NetAdapter                          {Disable-NetAdapter,...
Manifest   1.0.0.0    NetConnection                       {Get-NetConnectionPr...
Manifest   1.0.0.0    NetDiagnostics                      Get-NetView            
Manifest   1.0.0.0    NetEventPacketCapture               {New-NetEventSession...
Manifest   2.0.0.0    NetLbfo                             {Add-NetLbfoTeamMemb...
Manifest   1.0.0.0    NetNat                              {Get-NetNat, Get-Net...
Manifest   2.0.0.0    NetQos                              {Get-NetQosPolicy, S...
Manifest   2.0.0.0    NetSecurity                         {Get-DAPolicyChange,...
Manifest   1.0.0.0    NetSwitchTeam                       {New-NetSwitchTeam, ...
Manifest   1.0.0.0    NetTCPIP                            {Get-NetIPAddress, G...
Manifest   1.0.0.0    NetworkConnectivityStatus           {Get-DAConnectionSta...
Manifest   1.0.0.0    NetworkSwitchManager                {Disable-NetworkSwit...
Manifest   1.0.0.0    NetworkTransition                   {Add-NetIPHttpsCertB...
Manifest   1.0.0.0    PcsvDevice                          {Get-PcsvDevice, Sta...
Binary     1.0.0.0    PersistentMemory                    {Get-PmemDisk, Get-P...
Manifest   1.0.0.0    PKI                                 {Add-CertificateEnro...
Manifest   1.0.0.0    PnpDevice                           {Get-PnpDevice, Get-...
Manifest   1.1        PrintManagement                     {Add-Printer, Add-Pr...
Binary     1.0.11     ProcessMitigations                  {Get-ProcessMitigati...
Script     3.0        Provisioning                        {Install-Provisionin...
Manifest   1.1        PSDesiredStateConfiguration         {Set-DscLocalConfigu...
Script     1.0.0.0    PSDiagnostics                       {Disable-PSTrace, Di...
Binary     1.1.0.0    PSScheduledJob                      {New-JobTrigger, Add...
Manifest   2.0.0.0    PSWorkflow                          {New-PSWorkflowExecu...
Manifest   1.0.0.0    PSWorkflowUtility                   Invoke-AsWorkflow      
Manifest   1.0.0.0    ScheduledTasks                      {Get-ScheduledTask, ...
Manifest   2.0.0.0    SecureBoot                          {Confirm-SecureBootU...
Manifest   2.0.0.0    SmbShare                            {Get-SmbShare, Remov...
Manifest   2.0.0.0    SmbWitness                          {Get-SmbWitnessClien...
Manifest   1.0.0.0    StartLayout                         {Export-StartLayout,...
Manifest   2.0.0.0    Storage                             {Add-InitiatorIdToMa...
Manifest   1.0.0.0    StorageBusCache                     {Clear-StorageBusDis...
Manifest   2.0.0.0    TLS                                 {New-TlsSessionTicke...
Manifest   1.0.0.0    TroubleshootingPack                 {Get-Troubleshooting...
Manifest   2.0.0.0    TrustedPlatformModule               {Get-Tpm, Initialize...
Binary     2.1.639.0  UEV                                 {Clear-UevConfigurat...
Manifest   2.0.0.0    VpnClient                           {Add-VpnConnection, ...
Manifest   1.0.0.0    Wdac                                {Get-OdbcDriver, Set...
Manifest   2.0.0.0    Whea                                {Get-WheaMemoryPolic...
Manifest   1.0.0.0    WindowsDeveloperLicense             {Get-WindowsDevelope...
Script     1.0        WindowsErrorReporting               {Enable-WindowsError...
Manifest   1.0.0.0    WindowsSearch                       {Get-WindowsSearchSe...
Manifest   1.0.0.0    WindowsUpdate                       Get-WindowsUpdateLog   
Manifest   1.0.0.2    WindowsUpdateProvider               {Get-WUAVersion, Get...


    Directory: C:\Program Files (x86)\Microsoft SQL 
    Server\140\Tools\PowerShell\Modules


ModuleType Version    Name                                ExportedCommands       
---------- -------    ----                                ----------------       
Manifest   14.0       SQLPS                               {Backup-SqlDatabase,...

**Comment by markcowl** [comment]
@konstantinmds Thanks for the extra info - the issue could be with AzureRM and Az modules loaded in the same session, or something to do with your script.  Can you provide the details of your script, and also look at the loaded modules when this issue occurs (`Get-Module`)

**Comment by erich-wang** [comment]
@konstantinmds, I'm closing this issue since we have not gotten requested information from you for more than one month, please feel free to file a new issue if needed, thanks.
